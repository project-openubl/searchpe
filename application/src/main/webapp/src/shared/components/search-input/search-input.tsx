import React, { useState } from "react";

import { Button, InputGroup, TextInput } from "@patternfly/react-core";
import { SearchIcon } from "@patternfly/react-icons";

export interface SearchInputProps {
  onSearch: (filterText: string) => void;
}

export const SearchInput: React.FC<SearchInputProps> = ({ onSearch }) => {
  const [filterText, setFilterText] = useState("");

  const callSearch = () => {
    onSearch(filterText);
  };

  const handleOnKeyDown = (event: any) => {
    if (event.key === "Enter") {
      callSearch();
    }
  };

  return (
    <InputGroup>
      <TextInput
        name="filterText"
        type="search"
        aria-label="search input"
        value={filterText}
        onChange={setFilterText}
        onKeyDown={handleOnKeyDown}
        autoComplete="off"
      />
      <Button variant="control" aria-label="search button" onClick={callSearch}>
        <SearchIcon />
      </Button>
    </InputGroup>
  );
};
