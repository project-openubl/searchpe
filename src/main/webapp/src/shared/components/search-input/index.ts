export { SearchInput } from "./search-input";
