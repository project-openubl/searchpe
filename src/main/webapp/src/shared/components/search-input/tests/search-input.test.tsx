import React from "react";
import { mount, shallow } from "enzyme";

import { SearchInput } from "../search-input";
import { Button } from "@patternfly/react-core";

describe("SearchInput", () => {
  it("Renders without crashing", () => {
    const wrapper = shallow(<SearchInput onSearch={jest.fn} />);
    expect(wrapper).toMatchSnapshot();
  });

  it("Callback when click on search button", () => {
    const callbackSpy = jest.fn();
    const wrapper = mount(<SearchInput onSearch={callbackSpy} />);

    const nameInput = wrapper.find('input[name="filterText"]');
    (nameInput.getDOMNode() as HTMLInputElement).value = "myCompany";
    nameInput.simulate("change");
    wrapper.update();

    wrapper.find(Button).simulate("click");
    expect(callbackSpy).toHaveBeenCalledTimes(1);
    expect(callbackSpy).toHaveBeenCalledWith("myCompany");
  });

  it("Callback when click on press Enter", () => {
    const callbackSpy = jest.fn();
    const wrapper = mount(<SearchInput onSearch={callbackSpy} />);

    const nameInput = wrapper.find('input[name="filterText"]');
    (nameInput.getDOMNode() as HTMLInputElement).value = "myCompany";
    nameInput.simulate("change");
    nameInput.simulate("keydown", { key: "Enter" });
    wrapper.update();

    expect(callbackSpy).toHaveBeenCalledTimes(1);
    expect(callbackSpy).toHaveBeenCalledWith("myCompany");
  });
});
