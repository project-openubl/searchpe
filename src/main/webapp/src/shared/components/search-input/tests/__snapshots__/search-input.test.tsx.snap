// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SearchInput Renders without crashing 1`] = `
<InputGroup>
  <TextInput
    aria-label="search input"
    autoComplete="off"
    name="filterText"
    onChange={[Function]}
    onKeyDown={[Function]}
    type="search"
    value=""
  />
  <Button
    aria-label="search button"
    onClick={[Function]}
    variant="control"
  >
    <SearchIcon
      color="currentColor"
      noVerticalAlign={false}
      size="sm"
    />
  </Button>
</InputGroup>
`;
